### ISSUE TITLE
Truffle doesn't seem to compile a simple project on Linux

### ISSUE THREAD

**Issue by sarbogast** [trufflesuite/truffle#1798]
- [X] I've asked for help in the [Truffle Gitter](http://gitter.im/Consensys/truffle) before filing this issue.

---------------------------

## Issue

When I try to migrate [this simple project](
https://github.com/chainskills/greetings-truffle5) with Truffle 5.0.7 on Ubuntu Linux 18.04, the migrate command seems to return straigght to the prompt after showing "Compiling contracts...", without any error, and nothing gets compiled.

## Steps to Reproduce

I'm running Truffle 5.0.7 installed with `sudo npm install -g truffle` on Ubuntu 18.04 because it didn't work otherwise. 

I'm cloning [this simple project](
https://github.com/chainskills/greetings-truffle5).

Then I run Ganache v2.0.0-beta.2.

Then `truffle migrate --compile-all --reset --network ganache`.

## Expected Behavior

Compiling contracts and migrating them to Ganache.

## Actual Results

```
$ truffle migrate --network ganache

Compiling your contracts...
===========================
$
```

If I run the migrate command inside the truffle develop console, it simply stalls after showing that same line, without any error.

## Environment

* Operating System: Ubuntu 18.04
* Ethereum client: Ganache v2.0.0-beta.2
* Truffle version (`truffle version`): 5.0.7
* node version (`node --version`): 10.15.1
* npm version (`npm --version`): 6.4.1

**Comment by Bowfish** [comment]
If you have truffle installed as sudo you might experience different permission issues. It is recommended to install truffle as a normal user. Check [this](https://github.com/trufflesuite/truffle/issues/1711#issuecomment-464833806) post for further details. I had the same problem and could solve it with installing node and turffle as a normal user. The recommended way to install node is with nvm.

**Comment by sarbogast** [comment]
Indeed this worked. But forcing people to reinstall their whole development environment because Truffle or one of its dependencies doesn't work with NPM installed a certain way seems a bit extreme, doesn't it?

**Comment by gnidan** [comment]
@sarbogast wrote:
> Indeed this worked. But forcing people to reinstall their whole development environment because Truffle or one of its dependencies doesn't work with NPM installed a certain way seems a bit extreme, doesn't it?

Yep, you're 100% correct. My earlier assumption, considering that best practice recommends against `sudo npm`, was that it was not urgent to fix it, since it is arguably unsafe.

Earlier today, the rest of the team and I came to the conclusion that this reasoning is egregiously unsound. This is basically entirely thanks to all of the users whose bug reports have made it clear that it's **unexpectedly weird** and **basically unacceptable** for Truffle to fail to operate if installed by root.

@sarbogast your point here cuts directly to the heart of what was wrong with the prior assumption. Thank you for expressing it; I will be glad to remember it in the future when something else runs the risk of breaking user expectations like that.

Anyway, good news! Upon reaching the aforementioned conclusion, @eggplantzzz went and implemented what we believe to be a very straightforward fix! See #1801 for the PR if you're curious. I'm confident that we'll be able to get it into the next release, which will be sometime next week at the latest. (Thanks @eggplantzzz!)
